File: 10.math2/1978_check_prime_number.py
#!/usr/bin/env python


def check_prime_number(a):
    #print("\n{}".format(a))

    if a == 1: return False

    for i in range(2, a):
        #print(i)
        if (a//i)*i == a:
            return False

    return True


if __name__ == "__main__":
    n = int(input())
    nums = [int(i) for i in input().split()]
    cnt = 0

    for num in nums:
        if check_prime_number(num):
            cnt += 1

    print(cnt)

File: 10.math2/1978_check_prime_number.cpp
#include<iostream>

using namespace std;


bool check_prime_number(int a) {
    //cout << "\n" << a << endl;

    if (a == 1)
        return false;

    for (int i=2; i<a; ++i) {
        //cout << i << endl;
        if ((a/i)*i == a)
            return false;
    }

    return true;
}


int main() {
    int n, a;
    int cnt;

    cnt = 0;
    cin >> n;
    for (int i=0; i<n; ++i) {
        cin >> a;
        if (check_prime_number(a))
            cnt++;
    }

    cout << cnt << endl;

    return 0;
}
